### PR TITLE
Custom Dockerfile for CircleCI

### PR DESCRIPTION
**Pull request by eugene-babichenko**
**TBD**

- [x] Upload docker image to a registry (**who to ask?**: @NicolasDP or @mzabaluev )
- [ ] Add links to the uploaded image in CircleCI config
- [ ] Remove all of the tooling installation from the CircleCI config (already done in Dockerfile)
- [ ] Uncomment protoc related lines in https://github.com/input-output-hk/jormungandr/blob/master/jormungandr-integration-tests/build.rs

relates to #1020

### CODE REVIEW

**Comment by eugene-babichenko** [comment]
@NicolasDP where is our docker registry?

File: docker/ci/Dockerfile.nightly [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM rust:nightly
+RUN apt-get update && \
+    apt-get install -y protobuf-compiler && \
+    rustup component add rustfmt && \
+    cargo install cargo-audit
```
Comment (mzabaluev) [inline docker/ci/Dockerfile.nightly]: I'd add an apt-get cache cleanup to be really neat about the image size.

Comment (mzabaluev) [reply docker/ci/Dockerfile.nightly]: Also `cargo install` pulls loads of crates to compile, these caches should better be cleaned from the image as well.

We might want to build these images in CircleCI, where apt-get and cargo caches can be reused between runs like we do to speed up `cargo test` steps.

**Comment by mzabaluev** [comment]
Consider using [circleci/rust](https://hub.docker.com/r/circleci/rust) as the base for at least the stable image, as it has `rustfmt` installed and some other niceties that the CircleCI team thinks images should have in their pipelines.

File: docker/ci/Dockerfile.nightly [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM rustlang/rust:nightly
+
+LABEL org.label-schema.schema-version = "1.0"
+LABEL org.label-schema.vendor = "Input Output HK Limited"
+LABEL org.label-schema.description = "CI image for Jormungandr (nightly Rust)"
```
Comment (mzabaluev) [inline docker/ci/Dockerfile.nightly]: :+1:

**Comment by mzabaluev** [comment]
I guess we need to merge the branch with these Dockerfiles first, so we can set up Docker Hub builds for them and those should be set up against the master branch.

**Comment by NicolasDP** [comment]
What about : https://github.com/input-output-hk/jormungandr/tree/master/docker

**Comment by NicolasDP** [comment]
we have an existing docker directory and we could definitely reuse or at least concentrate docker images in the same directory, at least.

**Comment by eugene-babichenko** [comment]
I'm not sure about that. At least, I don't want to mix CI and production images in the same directory.

Or, if we do, it's better to make like that (now there is just `Dockerfile` in the root directory and `alpine` with `Dockerfile` in it):

```
circleci-nightly.dockerfile
circleci-stable.dockerfile
production.dockerfile
production-alpine.dockerfile
```

**Comment by eugene-babichenko** [comment]
@mzabaluev Do you mean automated builds? Then it makes sense to pin the exact versions of Rust and protoc in docker images and update them when we need to.

**Comment by NicolasDP** [comment]
what's the status on this?

**Pull request merged**